akka-stream/src/main/scala/akka/stream/impl/PhasedFusingActorMaterializer.scala

```diff
@@ -232,7 +237,7 @@ class IslandTracking(
       } else {
         if (Debug) println(s"    cross island forward wiring from port ${forwardWire.from} wired to local slot = $localInSlot")
         val publisher = forwardWire.phase.createPublisher(forwardWire.from, forwardWire.outStage)
-        currentPhase.takePublisher(localInSlot, publisher)
+        currentPhase.takePublisher(localInSlot, publisher, logic)
```

Comment:
that is the wrong logic there. There is a reason why the logic cannot be passed here.